src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -18,21 +18,44 @@ public class PackageSourceProvider : IPackageSourceProvider
 
         public PackageSourceProvider(
           ISettings settings)
-            : this(settings, ConfigurationDefaults.Instance.DefaultPackageSources)
+#pragma warning disable CS0618 // Type or member is obsolete
+            : this(settings, ConfigurationDefaults.Instance.DefaultPackageSources, enablePackageSourcesChangedEvent: true)
+#pragma warning restore CS0618 // Type or member is obsolete
         {
         }
 
+        [Obsolete("https://github.com/NuGet/Home/issues/8479")]
+        public PackageSourceProvider(
+          ISettings settings,
+          bool enablePackageSourcesChangedEvent)
+            : this(settings, ConfigurationDefaults.Instance.DefaultPackageSources, enablePackageSourcesChangedEvent)
+        {
+        }
+
+        public PackageSourceProvider(
```

Comment:
> Will it subscribe or not subscribe to events?

As a team, we need to decide. Basically, do we want to support detecting config file changes and auto-reload? Maybe we can do it just in the VS client code, and the packages don't need it. But until we decide, I don't know whether the events themselves are obsolete or not.

Reply:
>  Basically, do we want to support detecting config file changes and auto-reload? 

Eventually, but I don't necessarily think this would prevent us from doing that. 

It was just difficult for me to understand what codepath exactly we are trying to deprecate. 
Maybe you can elaborate on that?

Reply:
I want to deprecate the constructors that have a `enablePackageSourcesChangedEvent` parameter, or anything that calls the deprecated constructor.

In the long term, either the class always subscribes to the event, and the class is IDisposable. Or the class never subscribes to the event. In either case, the constructors with `enablePackageSourcesChangedEvent` will not be needed.

Reply:
Makes sense.